ekncontent/ekncontent/eknc-domain.c

```diff
@@ -0,0 +1,862 @@
+/* Copyright 2016 Endless Mobile, Inc. */
+
+#include "eknc-domain.h"
+
+#include "eknc-object-model.h"
+#include "eknc-utils.h"
+#include "eknc-utils-private.h"
+
+// Consolidate header would be nice
+#include <eos-shard/eos-shard-blob.h>
+#include <eos-shard/eos-shard-record.h>
+#include <eos-shard/eos-shard-dictionary.h>
+#include <eos-shard/eos-shard-shard-file.h>
+
+GQuark
+eknc_domain_error_quark (void)
+{
+  return g_quark_from_static_string ("eknc-domain-error-quark");
+}
+
+// This hash is derived from sha1('link-table'), and for now is the hardcoded
+// location of link tables for all shards.
+#define LINK_TABLE_ID "4dba9091495e8f277893e0d400e9e092f9f6f551"
+
+/**
+ * SECTION:domain
+ * @title: Domain
+ * @short_description: Grab content for a specific application
+ *
+ * The domain object handles querying for content for a specific application.
+ * Usually you will want to use #EkncEngine rather than use this directly.
+ */
+struct _EkncDomain
+{
+  GObject parent_instance;
+
+  gchar *app_id;
+  gchar *subscription_id;
+
+  EkncXapianBridge *xapian_bridge;
+  GFile *content_dir;
+  GSList *shards;
```

Comment:
optional, nitpick: I like putting in a comment here telling the type of the pointers stored in the `GSList`